kivy/graphics/vertex_instructions_line.pxi

```diff
@@ -1097,7 +1098,7 @@ cdef class Line(VertexInstruction):
 
     property rounded_rectangle:
         '''Use this property to build a rectangle, without calculating the
-        :attr:`points`. You can only set this property, not get it.
```

Comment:
Did you change this? If not then we need to retain this message.

Reply:
To perform the tests, it is necessary to access the resulting rounded rectangle, to know if it matches expectations.
Therefore, it is now possible to obtain the resulting rounded rectangle and to know, for example, if it was generated.

Another option would also be to keep this rectangle get unavailable for the user and create a specific method just to get the resulting rounded rectangle.
The first option seemed more interesting to me, as it would give more control to the user. But I would like to know your opinion on this.

Reply:
That's potentially useful indeed, but would be good to define what will be the result if the line was defined otherwise.

Reply:
1 - If the line does not fit within the requirements to be rendered, the rounded rectangle would be None (width or height less than 2 pixels).
2 - If the radius is set to 0, it will automatically be set to 1, to avoid internal issues with SmoothLine line construction. Furthermore, the difference between 0 and 1 would be almost imperceptible.
3 - And the maximum radius would be cut to half of the smallest dimension. It's the same behavior as RoundedRectangle. So if the smallest dimension is 100 and the user sets the radius to 200, the rounded rectangle will be generated with a radius of 50.

Would it be necessary to document this? I added a note regarding item 1, but I don't know if it would be necessary for the others. Since, for item 3, for example, there is no mention in the RoundedRectangle. And it's kind of a consensus that this is the expected behavior for high radius values in rounded rectangle shapes.

Reply:
1 - Sorry, i mean if the line was not defined as a rectagle (say, `Line(points=[0, 0, 100, 100])` then the user tries to read the property, or even if it was at some point defined as a rectangle (`line = Line(rounded_ectangle=[0, 0, 100, 100, 10, 10, 10, 10])`) but then later reset as something else (`line.ellipse = [0, 0, 100, 100]`) then the user tries to read the value, from the comment i infer that it would be `None` at least in the first case, but will it be reset to `None` in the second?
2 & 3 - i wasn't sure and so i just double checked, and it was previously possible to do such a shape 
![image](https://user-images.githubusercontent.com/22759/177210519-7a20a111-0581-41c4-b9ba-00442119dd77.png)
which is indeed not possible with `RoundedRectangle` and won't be possible with the new implementation.
Contrary to `RoundedRectangle` it's not possible to have a different radius per direction for each angle, so the APIs are not exactly similar (though that could be addressed in a different PR if someone wishes to make it so), but some people might have used the possibility of making the radius more than half the width/height, although it doesn't always make sense, it does in some cases.

Reply:
1 - With my last commits, it is now possible to get the generated shape for: `rounded_rectangle`, `rectangle`, `ellipse` and `circle`. Setting one of these properties will automatically reset the others to None, after the new shape parameters are determined.

An important note (not sure if it needs to be included in the documentation) is that the values obtained when trying to access a shape will not be immediately available, as with `points`, for example. The reason for this is that some shapes internally manage the radii, segments, and other parameters, according to the values passed. After that, the resulting shape is returned, which corresponds to the displayed shape.
Returning a value before that could, in some cases, create different values than what would actually be rendered. Even if the property value were updated later, the previous value would be inconsistent.
Of the new shapes that can be accessed, the only one that can be accessed in advance is `rectangle`. But a different logic just for this property, could be a little inconsistent. So I kept them all the same.

2 & 3 - The minimum radius value is still `1px`. This doesn't seem to make any difference from the radius `0`. But the maximum radius has been scaled up to the smallest dimension between width and height.
I addressed a bug that prevented the radius from being the maximum of the smallest width (minus `1px` of the min radius because of the rendering issues I mentioned above, of course). Now it's possible to get this shape from the image you shared, and it's still possible to get to a circle just passing values `greater than or equal to the smallest dimension` for all angles.

In general, fortunately Line and especially SmoothLine are much more reliable!